ReactSkia/modules/platform/libcurl/RSkNetworkingModule.cpp

```diff
@@ -266,16 +267,17 @@ void RSkNetworkingModule::sendProgressEventwrapper(NetworkRequest *networkReques
     if(ulnow >= ultotal)
       networkRequest->uploadComplete_ = true;
   }
-  if(networkRequest->downloadComplete_ == false && dltotal != 0) {
+  if(networkRequest->downloadComplete_ == false && dlnow != 0.000000) {
+    
     if(networkRequest->useIncrementalUpdates_) {
       if(networkRequest->responseType_ == "text") {
-        // TODO: responseText property needs to get and pass the value in the didReceiveNetworkIncrementalData event
-        sendEventWithName("didReceiveNetworkIncrementalData", folly::dynamic::array(networkRequest->requestId_ , "", dlnow,dltotal ));
+        std::scoped_lock lock(networkRequest->bufferLock_);
+        sendEventWithName("didReceiveNetworkIncrementalData", folly::dynamic::array(networkRequest->requestId_ ,&networkRequest->responseBuffer_[networkRequest->previousOffset_] , dlnow,dltotal ));
+        networkRequest->previousOffset_ =  networkRequest->responseBufferOffset_;
```

Comment:
in case of incrementalUpdates, we are holding the full responseBuffer .
Do we need to keep the full data ? Can we reset the buffer after sending the data .
This will avoid keeping the data which is not required .